Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -365,12 +377,9 @@ class Navigator {
 
     state.reset();
 
-    if (m_geometryVersion == GeometryVersion::Gen3) {
-      // Empirical pre-allocation of candidates for the next navigation
-      // iteration.
-      // @TODO: Make this user configurable through the configuration
-      state.stream.candidates().reserve(50);
-    }
+    // Empirical pre-allocation of candidates for the next navigation iteration.
+    // @TODO: Make this user configurable through the configuration
+    state.stream.candidates().reserve(50);
```

Comment:
```suggestion
    if (m_geometryVersion == GeometryVersion::Gen3) {
      // Empirical pre-allocation of candidates for the next navigation
      // iteration.
      // @TODO: Make this user configurable through the configuration
      state.stream.candidates().reserve(50);
    }
```
I think it would be good to keep the `if`